test/framework/controlplane_helpers.go

```diff
@@ -285,7 +285,7 @@ type UpgradeControlPlaneAndWaitForUpgradeInput struct {
 	Cluster                     *clusterv1.Cluster
 	ControlPlane                *controlplanev1.KubeadmControlPlane
 	KubernetesUpgradeVersion    string
-	UpgradeMachineTemplate      *string
+	UpgradeCPMachineTemplate    *string
```

Comment:
I think we can keep the previous field name here and in the MD func, as it's already clear that it's about CP/workers.

Reply:
sounds good, i will revert, thanks so much for the review and feedback

Reply:
No problem :)